### ISSUE TITLE
Invalid documentation with persistent services creation

### ISSUE THREAD

**Issue by migounette** [hashicorp/consul#4180]
### Operating system and Environment details
Linux (Redhat 7) - Version 1.0.7

Objective: Create a persistent service (eg. telegraf or consul)

After reading the section how to create a persistent services  based on the following: https://www.consul.io/docs/agent/services.html 

I found several issues, but in order to provide a Pull Request I need to help how to proceed:

The documentation says:
To configure a service, either provide it as a -config-file option to the agent or place it inside the -config-dir of the agent. The file must end in the .json or .hcl extension to be loaded by Consul. Check definitions can be updated by sending a SIGHUP to the agent. Alternatively, the service can be registered dynamically using the HTTP API.

Trying to use the example does not work, so REST API and static files are not aligned, moreover it's not possible to create a check section it needs to be dispatched on <datadir>/checks

Maybe a documenation update or a better paring in agent.go are needed, just let me know which is the best way.

I vote for 2, I mean align REST API and static files.

```
  "service": {
    "name": "redis", <== invalid
    "tags": ["primary"],
    "address": "",
    "meta": { <== invalid
      "meta": "for my service"
    }
    "port": 8000,
    "enable_tag_override": false,
    "checks": [
      {
        "args": ["/usr/local/bin/check_redis.py"],
        "interval": "10s"
      }
    ]
  }
}

```

Which seems to be coherent with agent.go file

Working:
```
{
  "service": {
    "service": "redis",
    "tags": ["primary"],
    "address": "",
    "port": 8000,
    "enable_tag_override": false
  }
}
```

**Comment by migounette** [comment]
@preetapan can you comment or point to some else
As written I write all necessary code for PR but I need instructions
Thanks

**Comment by pearkes** [comment]
@migounette it looks like from your example you caught invalid JSON that was in the documentation for a period of time and fixed in #4299. If not, please feel free to open a new issue.
